packages/lit-dev-content/site/blog/2022-02-07-eleventy.md

```diff
@@ -0,0 +1,176 @@
+```
+
+Then Eleventy will produce an HTML file `greeting/index.html` like this:
+
+```html
+<h1>Greetings</h1>
+
+<demo-greeter name="World">
+  <template shadowroot="open">
+    <style>
+      b { color: red; }
+    </style>
+    Hello <b>World</b>!
+  </template>
+</demo-greeter>
+```
+
+This HTML allows the browser to paint the initial state of the component's DOM
+and styles, preserving all of the encapsulation boundaries of web components,
+even before a single line of JavaScript has downloaded!
+
+Later, when the JavaScript definition of `<demo-greeter>` has finally loaded,
+the statically rendered component is upgraded to its fully interactive
+JavaScript implementation through a process called *hydration*.
+
+## Why should I use it?
+
+Statically rendering components can be an effective method for reducing the time
+it takes between a page starting to load, and the content becoming visible. This
+is an important consideration that contributes to the responsiveness of your
+site, particularly for users with slower connections and devices.
+
+By transmitting the initial state of your components as static HTML, the browser
+is able to paint the initial view much faster than it takes to download, parse,
+and evaluate the JavaScript implementations of those components.
+
+For a deeper dive into the benefits of static rendering, check out *[Rendering
+on the
+Web](https://developers.google.com/web/updates/2019/02/rendering-on-the-web)*,
+and for a discussion of how to measure this aspect of web performance, check out
+*[Largest Contentful Paint (LCP)](https://web.dev/lcp/)*.
+
+## How does it work?
+
+### Lit Labs SSR
+
+The plugin uses `@lit-labs/ssr` under the hood, which works by creating an
+instance of each component in a lightweight browser-like environment within the
+Eleventy Node process, and evaluating the element's template using a special
+version of Lit's `render` function. This special `render` function behaves much
+like the `render` function that Lit uses in the browser, except that it emits a
+stream of strings instead of writing to the DOM, allowing it to be efficiently
+written out as static HTML.
+
+For more information about `@lit-labs/ssr`, check out its [GitHub
+repo](https://github.com/lit/lit/tree/main/packages/labs/ssr). We'll be
+expanding the documentation soon to help developers integrate `@lit-labs/ssr`
+into more tools and frameworks!
+
+### Declarative Shadow DOM
+
+When `@lit-labs/ssr` renders a component, it uses `<template shadowroot>` HTML
+elements to represent the component's shadow DOM and styles. When the browser
+HTML parser encounters a `<template shadowroot>`, it applies that template's
+content as the shadow root of its parent element. This way, all of the DOM and
+style encapsulation guarantees of web components are preserved.
+
+As of February 2022, Chrome and Edge have native support for Declarative Shadow
+DOM, but Firefox and Safari have not yet implemented it. A very small polyfill
```

Comment:
Doesn't need action, but "polyfill" is used here whilst the readme uses "ponyfill".

Maybe change all usages to be "ponyfill"? [Is this the definition for ponyfill?](https://github.com/sindresorhus/ponyfill).

🦄🦄🦄🦄🦄🦄🦄🦄🦄

Reply:
Yeah, the distinction is that a polyfill patches an API into existence, whereas a ponyfill serves exposes an equivalent API but through a separate explicit API. The declarative template shadow root is a ponyfill by this definition, because you have to call a `hydrateShadowRoots` function -- it's not enough to just import the script.

I think the distinction is not super useful here though, plus I think we might switch it to be a proper polyfill sometime soon.

Changed the terminology in the README at https://github.com/lit/lit/pull/2499, and the one spot here too.